### PR TITLE
Bugfixes/rordev 130

### PR DESCRIPTION
**Pull request by coutoPL**
Improvement (ES) more reliable in-index settings loading of ES with ROR startup
Fix (ES) _cat/indices API bug fixes
Fix (ES) _cat/templates API return empty list instead of FORBIDDEN when no indices are found

### CODE REVIEW

File: core/src/main/scala/tech/beshu/ror/utils/ScalaOps.scala [hunk 1]
```diff
@@ -35,6 +36,16 @@ object ScalaOps {
...
   }
 
+  implicit class ArrayOps[T : ClassTag](val array: Array[T]) {
+    def asSafeSet: Set[T] = {
+      Option(array).getOrElse(Array.empty[T]).toSet
+    }
+  }
+
+  implicit class SetOps[T](val value: T) extends AnyVal {
+    def asSafeSet: Set[T] = Option(value).map(Set(_)).getOrElse(Set.empty)
```
Comment (clutroth) [inline core/src/main/scala/tech/beshu/ror/utils/ScalaOps.scala]:
```scala
Option(value).toSet
```

File: integration-tests/src/test/eshome/config/readonlyrest.yml [hunk 2]
```diff
@@ -13,22 +13,15 @@ readonlyrest:
       type: allow
```
Comment (clutroth) [inline integration-tests/src/test/eshome/config/readonlyrest.yml]: rm integration-tests/src/test/eshome/config/elasticsearch.keystore

Comment (clutroth) [reply integration-tests/src/test/eshome/config/readonlyrest.yml]: and ignore

File: core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/BlockContext.scala [hunk 3]
```diff
@@ -74,6 +75,13 @@ object BlockContext {
   object Outcome {
     final case class Exist[T](value: T) extends Outcome[T]
```
Comment (clutroth) [inline core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/BlockContext.scala]: Isn't it `Option`?

File: es51x/src/main/scala/tech/beshu/ror/es/request/regular/RegularRequestHandler.scala [hunk 4]
```diff
@@ -88,26 +93,58 @@ class RegularRequestHandler[Request <: ActionRequest, Response <: ActionResponse
...
+        requestInfo.writeSnapshots(BlockContextRawDataHelper.snapshotsFrom(blockContext))
+        requestInfo.writeRepositories(BlockContextRawDataHelper.repositoriesFrom(blockContext))
+        writeCommonParts(requestInfo, blockContext)
+
+        proceed(searchListener)
+    }
+  }
 
-    proceed(searchListener)
+  private def applyIfExists[T](outcome: Outcome[T])(function: T => Unit): Unit = {
```
Comment (clutroth) [inline es51x/src/main/scala/tech/beshu/ror/es/request/regular/RegularRequestHandler.scala]: haven't you defined functor for this?

File: es51x/src/main/scala/tech/beshu/ror/es/request/regular/RegularRequestHandler.scala [hunk 5]
```diff
@@ -88,26 +93,58 @@ class RegularRequestHandler[Request <: ActionRequest, Response <: ActionResponse
...
+      case Outcome.NotExist =>
+    }
+  }
+
+  private def writeCommonParts(requestInfo: RequestInfo, blockContext: BlockContext): Unit = {
+    requestInfo.writeResponseHeaders(BlockContextRawDataHelper.responseHeadersFrom(blockContext))
+    requestInfo.writeToThreadContextHeaders(BlockContextRawDataHelper.contextHeadersFrom(blockContext))
+  }
+
+  private def emptySetOfFoundIndices(blockContext: BlockContext) = {
```
Comment (clutroth) [inline es51x/src/main/scala/tech/beshu/ror/es/request/regular/RegularRequestHandler.scala]: this too looks's like doubled `Option.forall(_.isEmpty)`, or something like tkis

File: core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/BlockContext.scala (hunk 3 above)
Comment (coutoPL) [reply core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/BlockContext.scala]: semantics is the same, but IMO type alias is not enough. now, IMO, is readable and descriptive

**Pull request merged**